server/app/services/docker/service_creator.rb

```diff
@@ -173,6 +175,46 @@ def build_networks
       networks
     end
 
+    def build_volumes(instance_number)
+      volume_specs = []
+      grid_service.volumes.each do |vol|
+        vol_string = remove_volume_flags(vol.dup)
+        elements = vol_string.split(':')
+        host_path_or_vol = elements[-2]
+        if host_path_or_vol.nil? || host_path_or_vol.start_with?('/')
+          next
+        end
+        container_path = elements[-2]
+        volume =
+          grid_service.stack.volumes.find_by(name: host_path_or_vol) ||
+          grid_service.stack.external_volumes.find_by(name: host_path_or_vol).volume
+          # TODO Need o handle nils? Should not be possible since the stack is invalid if no volumes found
+        if volume
+          volume_name = volume.name_for_service(grid_service, instance_number)
+          volume_specs << {
+              name: volume_name,
+              driver: volume.driver,
+              scope: volume.scope,
+              driver_opts: volume.driver_opts
+          }
+          vol.sub!(host_path_or_vol, volume_name)
```

Comment:
Ew, string mutation, this is super scary, I don't like it. Please explicitly `map` the `spec[:volumes]`.